cpp/arcticdb/processing/aggregation.cpp

```diff
@@ -161,4 +161,41 @@ SegmentInMemory MeanAggregatorData::finalize(const ColumnName& output_column_nam
     }
     return res;
 }
+
+void CountAggregatorData::aggregate(const std::optional<ColumnWithStrings>& input_column, const std::vector<size_t>& groups, size_t unique_values) {
+    if(input_column.has_value()) {
+        input_column->column_->type().visit_tag([&] (auto type_desc_tag) {
+            using TypeDescriptorTag =  decltype(type_desc_tag);
+            using RawType = typename TypeDescriptorTag::DataTypeTag::raw_type;
+
+            aggregated_.resize(unique_values);
+
+            auto col_data = input_column->column_->data();
+            while (auto block = col_data.next<TypeDescriptorTag>()) {
+                auto ptr = reinterpret_cast<const RawType *>(block.value().data());
+                for (auto i = 0u; i < block.value().row_count(); ++i) {
+                    if(!std::isnan(static_cast<double>(ptr[i]))) {
```

Comment:
We can use `if constexpr(is_floating_point_type(TypeDescriptorTag::date_type))` to avoid doing the `NaN` check for integer columns.

Reply:
In fact, the result only depends on `col_data` at all for float and double columns. For integers and strings you can just use the `groups` vector.